packages/editor/src/store/actions.js

```diff
@@ -432,175 +380,48 @@ export function __experimentalOptimisticUpdatePost( edits ) {
+		content: yield select( 'core/editor', 'getEditedPostContent' ),
+	} );
 
-	// Optimistically apply updates under the assumption that the post
-	// will be updated. See below logic in success resolution for revert
-	// if the autosave is applied as a revision.
+	yield __experimentalRequestPostUpdateStart( options );
+	const postType = yield select( 'core/editor', 'getCurrentPostType' );
+	const postId = yield select( 'core/editor', 'getCurrentPostId' );
 	yield dispatch(
-		STORE_KEY,
-		'__experimentalOptimisticUpdatePost',
-		toSend
-	);
-
-	let path = `/wp/v2/${ postType.rest_base }/${ post.id }`;
-	let method = 'PUT';
-	if ( isAutosave ) {
-		const currentUser = yield select( 'core', 'getCurrentUser' );
-		const currentUserId = currentUser ? currentUser.id : undefined;
-		const autosavePost = yield select( 'core', 'getAutosave', post.type, post.id, currentUserId );
-		const mappedAutosavePost = mapValues( pick( autosavePost, AUTOSAVE_PROPERTIES ), getPostRawValue );
-
-		// Ensure autosaves contain all expected fields, using autosave or
-		// post values as fallback if not otherwise included in edits.
-		toSend = {
-			...pick( post, AUTOSAVE_PROPERTIES ),
-			...mappedAutosavePost,
-			...toSend,
-		};
-		path += '/autosaves';
-		method = 'POST';
-	} else {
-		yield dispatch(
-			'core/notices',
-			'removeNotice',
-			SAVE_POST_NOTICE_ID
-		);
-		yield dispatch(
-			'core/notices',
-			'removeNotice',
-			'autosave-exists'
-		);
-	}
-
-	try {
-		const newPost = yield apiFetch( {
-			path,
-			method,
-			data: toSend,
-		} );
-
-		if ( isAutosave ) {
-			yield dispatch( 'core', 'receiveAutosaves', post.id, newPost );
-		} else {
-			yield dispatch( STORE_KEY, 'resetPost', newPost );
-		}
-
-		yield dispatch(
-			STORE_KEY,
-			'__experimentalRequestPostUpdateSuccess',
-			{
-				previousPost: post,
-				post: newPost,
-				options,
-				postType,
-				// An autosave may be processed by the server as a regular save
-				// when its update is requested by the author and the post was
-				// draft or auto-draft.
-				isRevision: newPost.id !== post.id,
-			}
-		);
-
-		const notifySuccessArgs = getNotificationArgumentsForSaveSuccess( {
-			previousPost: post,
-			post: newPost,
-			postType,
-			options,
-		} );
-		if ( notifySuccessArgs.length > 0 ) {
-			yield dispatch(
-				'core/notices',
-				'createSuccessNotice',
-				...notifySuccessArgs
-			);
-		}
-	} catch ( error ) {
-		yield dispatch(
-			STORE_KEY,
-			'__experimentalRequestPostUpdateFailure',
-			{ post, edits, error, options }
-		);
-		const notifyFailArgs = getNotificationArgumentsForSaveFail( {
-			post,
-			edits,
-			error,
-		} );
-		if ( notifyFailArgs.length > 0 ) {
-			yield dispatch(
-				'core/notices',
-				'createErrorNotice',
-				...notifyFailArgs
-			);
+		'core',
+		'saveEditedEntityRecord',
+		'postType',
+		postType,
+		postId,
+		{
+			...options,
+			getSuccessNoticeActionArgs: ( previousEntity, entity, type ) => {
```

Comment:
Can't we use selectors after the "save" action finishes in order to trigger these notices here instead?

Reply:
No, because nothing guarantees that the micro-tasks triggered in that execution context will run before control returns here. I tried and ended up with the notices firing before the saving actually finished.